doc/website/blog/2021-09-10-openapi-generator.md

```diff
@@ -0,0 +1,329 @@
+learn more about it, how to connect to different types of databases, run migrations or work with entities head over to
+the [Setup Tutorial](https://entgo.io/docs/tutorial-setup/).
+
+### Generate an OAS file
+
+The first step on our way to the OAS file is to create an ent schema graph:
+
+```shell
+go run -mod=mod entgo.io/ent/cmd/ent init Fridge Compartment Content
+```
+
+I have multiple fridges with multiple compartments, and me and my SO want to know its contents at all times. We will
+create a Go server to hold the state and with the generated OAS file you are invited to build a frontend to manage
+fridges and contents in the language or your choice by using the Swagger Codegen. You can find a `generate.go` that uses
+docker to generate a client [here](https://github.com/masseelch/elk/blob/master/internal/openapi/ent/generate.go).
+
+Let's create our schema:
+
+```go title="ent/fridge.go"
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+)
+
+// Fridge holds the schema definition for the Fridge entity.
+type Fridge struct {
+	ent.Schema
+}
+
+// Fields of the Fridge.
+func (Fridge) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title"),
+	}
+}
+
+// Edges of the Fridge.
+func (Fridge) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("compartments", Compartment.Type),
+	}
+}
+```
+
+```go title="ent/compartment.go"package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+)
+
+// Compartment holds the schema definition for the Compartment entity.
+type Compartment struct {
+	ent.Schema
+}
+
+// Fields of the Compartment.
+func (Compartment) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Compartment.
+func (Compartment) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("fridge", Fridge.Type).
+			Ref("compartments").
+			Unique(),
+		edge.To("contents", Content.Type),
+	}
+}
+```
+
+```go title="ent/content.go"package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+)
+
+// Content holds the schema definition for the Content entity.
+type Content struct {
+	ent.Schema
+}
+
+// Fields of the Content.
+func (Content) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Content.
+func (Content) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("compartment", Compartment.Type).
+			Ref("contents").
+			Unique(),
+	}
+}
+```
+
+Now, let's generate the Ent code and the OAS file.
+
+```shell
+go generate ./...
+```
+
+In addition to the files Ent normally generates, another file names `openapi.json` has been created. Copy its contents
+and paste them into the [Swagger Editor](https://editor.swagger.io/), You should see three groups: Compartment, Content
+and Fridge. If you happen to open up the POST operation tab in the Fridge group, you see a description of the expected
```

Comment:
```suggestion
and paste them into the [Swagger Editor](https://editor.swagger.io/), You should see three groups: `Compartment`, `Content`
and `Fridge`. If you happen to open up the POST operation tab in the Fridge group, you see a description of the expected
```

I think groups should be highlighted somehow.

Also, maybe add a screenshot from the Swagger editor?

Reply:
Agree to the highliting, but I think making them bold/strong is fits more than backticks.

I will add some screenshots to the post.